components/config/prefs.rs

```diff
@@ -2,659 +2,384 @@
+    pub js_throw_on_debuggee_would_run: bool,
+    pub js_timers_minimum_duration: i64,
+    pub js_wasm_baseline_enabled: bool,
+    pub js_wasm_enabled: bool,
+    pub js_wasm_ion_enabled: bool,
+    pub js_werror_enabled: bool,
+    pub layout_animations_test_enabled: bool,
+    pub layout_columns_enabled: bool,
+    pub layout_grid_enabled: bool,
+    pub layout_container_queries_enabled: bool,
+    pub layout_css_transition_behavior_enabled: bool,
+    pub layout_flexbox_enabled: bool,
+    pub layout_legacy_layout: bool,
+    pub layout_threads: i64,
+    pub layout_unimplemented: bool,
+    pub layout_writing_mode_enabled: bool,
+    /// Enable hardware acceleration for video playback.
+    pub media_glvideo_enabled: bool,
+    /// Enable a non-standard event handler for verifying behavior of media elements during tests.
+    pub media_testing_enabled: bool,
+    pub network_enforce_tls_enabled: bool,
+    pub network_enforce_tls_localhost: bool,
+    pub network_enforce_tls_onion: bool,
+    pub network_http_cache_disabled: bool,
+    pub network_local_directory_listing_enabled: bool,
+    pub network_mime_sniff: bool,
+    /// Ignore `std::io::Error` with `ErrorKind::UnexpectedEof` received when a TLS connection
+    /// is closed without a close_notify.
+    ///
+    /// Used for tests because WPT server doesn't properly close the TLS connection.
+    // TODO: remove this when WPT server is updated to use a proper TLS implementation.
+    pub network_tls_ignore_unexpected_eof: bool,
+    pub session_history_max_length: i64,
+    /// The background color of shell's viewport. This will be used by OpenGL's `glClearColor`.
+    pub shell_background_color_rgba: [f64; 4],
+    pub webgl_testing_context_creation_error: bool,
+    /// Number of workers per threadpool, if we fail to detect how much
+    /// parallelism is available at runtime.
+    pub threadpools_fallback_worker_num: i64,
+    /// Maximum number of workers for the Image Cache thread pool
+    pub threadpools_image_cache_workers_max: i64,
+    /// Maximum number of workers for the Networking async runtime thread pool
+    pub threadpools_async_runtime_workers_max: i64,
+    /// Maximum number of workers for the Core Resource Manager
+    pub threadpools_resource_workers_max: i64,
+    /// Maximum number of workers for webrender
+    pub threadpools_webrender_workers_max: i64,
 }
 
-impl TryFrom<&PrefValue> for StyloPrefValue {
-    type Error = TryFromPrefValueError;
-
-    fn try_from(value: &PrefValue) -> Result<Self, Self::Error> {
-        match *value {
-            PrefValue::Int(value) => {
-                if let Ok(value) = value.try_into() {
-                    Ok(Self::Int(value))
-                } else {
-                    Err(TryFromPrefValueError::IntegerOverflow(value))
-                }
-            },
-            PrefValue::Bool(value) => Ok(Self::Bool(value)),
-            _ => Err(TryFromPrefValueError::UnmappedType),
+impl Preferences {
```

Comment:
This is the new way that preferences are set by default. I've taking these values from `resources/prefs.json`. In some cases (like for `layout_threads`) a default was defined in Servo but overridden by the resource file. I've used the one from the JSON in those cases.